### PR TITLE
Add bentoctl to BentoML docs and example

### PR DESCRIPTION
**Pull request by safoinme**
## Describe changes
Add Bentoctl to docs and example.
Fix a type error in the bento_build step

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [X] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/component-gallery/model-deployers/bentoml.md [hunk 1]
```diff
@@ -38,10 +38,16 @@ you can take a look to one of the other
...
+The BentoML also allows you to deploy your models in a more complex
+production grade dedicated ways, [bentoctl](https://github.com/bentoml/bentoctl)
+is one of this ways, bentoctl takes your built Bento from a ZenML
+pipeline and deploy it with bentoctl in a cloud environment such as AWS Lambda, 
+AWS SageMaker, Google Cloud Functions, Google Cloud AI Platform, Azure Functions.
+Read more about this in the [From Local to Cloud with bentoctl section](#from-local-to-cloud-with-bentoml).
+
+Note: The bentoctl integration implementation is still in progress and will be
+available soon. The integration will allow you to deploy your models to a specific
+cloud provider with just few lines of code using ZenML built-in steps.
```
Comment (AlexejPenner) [inline docs/book/component-gallery/model-deployers/bentoml.md:50]:
```suggestion
BentoML also allows you to deploy your models in a more complex
production grade setting. [Bentoctl](https://github.com/bentoml/bentoctl)
is one of the tools that can help you get there. Bentoctl takes your built Bento from a ZenML
pipeline and deploys it with bentoctl into a cloud environment such as AWS Lambda, 
AWS SageMaker, Google Cloud Functions, Google Cloud AI Platform or Azure Functions.
Read more about this in the [From Local to Cloud with bentoctl section](#from-local-to-cloud-with-bentoml).

Note: The bentoctl integration implementation is still in progress and will be
available soon. The integration will allow you to deploy your models to a specific
cloud provider with just few lines of code using ZenML built-in steps.
```

File: docs/book/component-gallery/model-deployers/bentoml.md [hunk 2]
```diff
@@ -214,6 +220,55 @@ def predictor(
         rich_print(f"Prediction for {img} is {result}")
 ```
 
+### From Local to Cloud with bentoctl
+
+bentoctl helps deploy any machine learning models as production-ready API 
+endpoints on the cloud. It is a command line tool that provides a simple
+interface to manage your BentoML bundles.
```
Comment (AlexejPenner) [inline docs/book/component-gallery/model-deployers/bentoml.md:227]:
```suggestion
Bentoctl helps deploy any machine learning models as production-ready API 
endpoints into the cloud. It is a command line tool that provides a simple
interface to manage your BentoML bundles. 
```

**Review by AlexejPenner — CHANGES_REQUESTED** [review]
Slight formulation changes, otherwise you make the bentoml production path sound easy :)

File: docs/book/component-gallery/model-deployers/bentoml.md [hunk 3]
```diff
@@ -214,6 +220,55 @@ def predictor(
...
 ```
 
+### From Local to Cloud with bentoctl
+
+Bentoctl helps deploy any machine learning models as production-ready API 
+endpoints into the cloud. It is a command line tool that provides a simple
+interface to manage your BentoML bundles. 
+
+The bentoctl CLI provides list of opetors which are plugins that interact with 
+cloud services, some of these operators are:
```
Comment (htahir1) [inline docs/book/component-gallery/model-deployers/bentoml.md:230]:
```suggestion
The bentoctl CLI provides list of operators which are plugins that interact with cloud services, some of these operators are:
```

File: examples/bentoml_deployment/README.md [hunk 4]
```diff
@@ -254,6 +342,12 @@ To stop any prediction servers running in the background, use the
 zenml model-deployer models delete cd38d6e6-467b-46e0-be13-3112c6e65d0e
 ```
 
+To delete the Sagemaker deployment, run the following command:
+
+```shell
+benctl destroy
```
Comment (htahir1) [inline examples/bentoml_deployment/README.md:348]: benctl?

**Review by htahir1 — CHANGES_REQUESTED** [review]
Left two RC's but there are some things which are still unclear to me:

* What happens on remote orchestrators?
* Do you need to run the bento builder step only? Or also bento deployer?
* What is the recommended path to production? Run a pipeline with bentoctl builder and then run the operator commands? Does each pipeline run build a new bento each time and you need the version each time?

**Comment by safoinme** [comment]
@htahir1 Regarding your questions:
1- At the moment the bentoml deployment can't be done using a remote orchestrator for two main reasons (1- deployed service will be terminated with the step end. 2- `bentoctl` functionalities work with CLI which needs to be provided as some sort of parameters pre-run and any remote deployment will require the right cloud credentials + bentoctl build a docker image in the process which also must be done pre-run or it will be a mess due to `docker in docker` complexity) 
However, the bento_build step can be done on a remote orchestrator, and then the user can build a custom pipeline that uses post_execution to get the built bento to deploy it locally or use bentoctl on it. (these challenges are what we will try to solve with the bentoctl integration to allow remote cloud deployment if possible)

2- You only need the bento_builder step to use `bentoctl` so we can say that it's just an alternative to bento_deployer which will become `bento_deployer_local` and `bento_deployer_cloud` once the integration of bentoctl is done.

3- It depends on the problem and training pipeline, but I would say have to have the bento_builder in the same pipeline as training (for now) because loading back a big model that maybe was trained in a GPU environment and can be a bit off performance issue and extra copying of the same files (model) in and to storage with both pipelines. The last thing is you don't need actually the version part the name is enough unless you want to be too specific, as bentoctl picks up the latest one by default.

File: docs/book/component-gallery/model-deployers/bentoml.md [hunk 5]
```diff
@@ -214,6 +220,55 @@ def predictor(
...
+# Build and push the docker image to the cloud
+bentoctl build -b $BENTO_TAG -f deployment_config.yaml
+
+# Deploy to the cloud
+bentoctl apply -f deployment_config.yaml
+```
+
+deployment_name = "zenml-bentoml-example"
+region = "eu-central-1"
+instance_type = "ml.m5.large"
```
Comment (fa9r) [inline docs/book/component-gallery/model-deployers/bentoml.md:269]: Should this really be here?

File: .pyspelling-ignore-words [hunk 6]
```diff
@@ -938,6 +938,7 @@ dec
 decodable
 deepchecks
 defaultdict
+DEPLOYEMNT
```
[the remark this answers is not in the document]
Comment (fa9r) [reply .pyspelling-ignore-words:941]: probably you don't need to ignore this if you fix it below ;)

**Review by fa9r — APPROVED** [review]

**Review by htahir1 — APPROVED** [review]

**Pull request merged**